### ISSUE TITLE
False positive with misrefactored_assign_op

### ISSUE THREAD

**Issue by llogiq** [rust-lang-nursery/rust-clippy#1239]
```
411 |         v *= v * v;
    |         ^^^^^^^^^^
    |
help: replace it with
    |         v *= v;
```

There is obviously a difference between the original and the suggestion.

**Comment by oli-obk** [comment]
The assumption is that noone wants to write v *= v * v because thats not trivial to comprehend

**Comment by llogiq** [comment]
Even then the suggestion changes behavior. It may be easier to comprehemd, but it's *wrong*.

**Comment by pengowen123** [comment]
The lint assumes that `v *= v * v` is likely a mistake, and suggests `v *= v` instead. It is not meant to suggest an equivalent, but rather to catch mistakes. What should the lint do in this case?

**Comment by llogiq** [comment]
Ask if `v = v * v` or `v = v * v * v` is meant, instead of arbitrarily guessing.

**Comment by mcarton** [comment]
👍 I think that a good approach.
